aten/src/ATen/native/cpu/group_norm_kernel.cpp

```diff
@@ -640,10 +798,20 @@ void GroupNormBackwardKernelImpl(
     Tensor& dX,
     Tensor& dgamma,
     Tensor& dbeta) {
+  const bool mixed_type = is_mixed_type(dY, gamma);
   AT_DISPATCH_FLOATING_TYPES_AND(
       ScalarType::BFloat16, X.scalar_type(), "GroupNormBackwardKernelImpl", [&]() {
-        GroupNormBackwardKernelImplInternal<scalar_t>(
+      // In training, using Amp to enable BFloat16 is recommended.
+      // It will keep module parameters in acc dtype i.e. float
+      // while input/output will be in BFloat16.
+      // Using parameters in BFloat16 will cause high precision loss.
+      if(mixed_type) {
```

Comment:
Same feedback as for https://github.com/pytorch/pytorch/pull/81852 : please consider who this could be constrained to just `BFloat16`, perhaps it could be handled outside of the generic switch statement

Reply:
Fixed in https://github.com/pytorch/pytorch/pull/89485